pkg/api/types.go

```diff
@@ -1192,13 +1195,15 @@ type ServiceSpec struct {
 	// None can be specified for headless services when proxying is not required
 	ClusterIP string `json:"clusterIP,omitempty"`
 
-	// Type determines how the service will be exposed.  Valid options: ClusterIP, NodePort, LoadBalancer
-	Type ServiceType `json:"type,omitempty"`
 
 	// ExternalIPs are used by external load balancers, or can be set by
 	// users to handle external traffic that arrives at a node.
 	ExternalIPs []string `json:"externalIPs,omitempty"`
 
+	// Only applies to Service Type: LoadBalancer
+	// LoadBalancer will get created with the IP specified in this field.
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
```

Comment:
I'm slightly confused by what LoadBalancerIP actually is :/

I thought it was some idea of "chassis IP", which OpenStack doesn't have.  You _can_ refer to loadbalancers by an *ID* however.  In some deployments these refer to real hardware loadbalancers, so can't be created/destroyed by regular users.  In LBaaS v2 API, each "loadbalancer" can only have a single external IP (aka VIP), so exposing this might not mesh well with `ExternalIPs[]`.

Your description above "allocate me an IP that is mine until I release it" sounds like you might be referring to something OpenStack calls a "floating IP".  Yes, you can have a public IP allocated from a pool - and associate that with a loadbalancer.  And you can pre-allocate the floating-ip before configuring any loadbalancer functionality.

I'm still struggling to understand exactly how `LoadBalancerIP` is intended to differ from `ExternalIPs`.  We might indeed be talking about some common functionality, but at the moment neither the code comment here, the commit description, nor the discussion so far has helped me understand the expected semantics beyond "it's an IP that has something to do with the loadbalancer" :P